tests/php/Forms/GridField/GridFieldTest.php

```diff
@@ -716,4 +732,86 @@ public function testAddRequestToURL()
             '/class-name/item/1?gridState-Test%5BState%5D%5BColumn%5D=Name'
         );
     }
+
+    public static function provideSudoModeShowsCollapsedField(): array
+    {
+        return [
+            'form-protected-do-protected-sudo-active' => [
+                'formProtected' => true,
+                'dataObjectProtected' => true,
+                'sudoActive' => true,
+                'expected' => false,
+            ],
+            'form-protected-do-protected-sudo-inactive' => [
+                'formProtected' => true,
+                'dataObjectProtected' => true,
+                'sudoActive' => false,
+                'expected' => true,
+            ],
```

Comment:
This is `true` as in we should expect the component, but it doesn't test whether the actual sudo mode field is rendered or not like the test method name implies.
The component should be present, but the field shouldn't be rendered in this scenario because the form is protected.

Reply:
Have updated test to check rendered HTML instead